frontend/src/app/auth.service.ts

```diff
@@ -36,6 +36,16 @@ export class AuthService {
       });
   }
 
+  logout() {
+      this.http.get(this._logoutUrl, { withCredentials: true })
+        .subscribe(rsp => {
+          if (this._currentUser) {
+            this._currentUser = undefined;
+            this.changeUserEvent.emit(this._currentUser);
```

Comment:
Overall, emitting the events for any other purpose than communication between parent / child component should generally be discouraged.

Whichever component is interested in the user being logged in, it should get this information via the Observable subscription.

So, I would suggest:
1. No event here (and in any other use case like this one).
2. Expose the current user from this service via the Observable (you can probably utilize `ReplaySubject` with `asObservable` somewhere in the way.
3. Subscribe to this observable in the `ngOnInit` method of the components that might need it.